src/ontology/uberon.Makefile

```diff
@@ -1282,6 +1282,24 @@ ifeq ($(strip $(IMP)),true)
 mappings/%-mappings.sssom.tsv: %-mappings
 	wget -O $@ http://purl.obolibrary.org/obo/$*/$*-mappings.sssom.tsv
 
+# Special case for the Biomappings set which is provided with the
+# metadata in a separate file. We assemble a single YAML+TSV file and
+# we also filter out anything that has nothing to do with Uberon: the
+# set is quite large and we commit it to the repository, so we want to
+# keep it to the minimum stuff we actually need.
+# (We do the filtering with sed because "sssom dosql" is WAY too slow
+# for what it does.)
+mappings/biomappings-mappings.sssom.tsv: $(TMPDIR)/biomappings.sssom.tsv \
+					 $(TMPDIR)/biomappings.sssom.yml
+	sed 's/^/#/' $(TMPDIR)/biomappings.sssom.yml > $@
+	sed -n '1p; /UBERON:/p' $(TMPDIR)/biomappings.sssom.tsv >> $@
```

Comment:
No, `sssom-py` is already included in the ODK anyway, so that is not the problem.

The problem is that `sssom-py` is just excruciatingly slow, sorry. Just assembling a single YAML+TSV file (which is necessary because `sssom dosql` cannot use a mapping set with non-embedded metadata) takes (on my machine) 20 seconds:

```sh
$ time sssom parse -I tsv biomappings.sssom.tsv -m biomappings.sssom.yml -o combined-biomappings.sssom.tsv
21.57s user 0.25s system 107% cpu 20.334 total
```

Then filtering out everything that is not relevant for Uberon takes 30 more seconds:

```sh
$ time sssom dosql -Q "SELECT * FROM df1 WHERE (subject_id LIKE 'UBERON:%' OR object_id LIKE 'UBERON:%')" -o filtered.sssom.tsv combined.sssom.tsv
28.24s user 0.68s system 108% cpu 27.409 total
```

`sed` does that in less than one second.

In the future, when the next release of the ODK (which includes SSSOM-Java’s `sssom-cli`), we will be able to do everything in a single step which will also take less than one second:

```sh
time sssom-cli --prefix=UBERON=http://purl.obolibrary.org/obo/UBERON_ -R "(subject==UBERON:* || object==UBERON:*) -> include()" -i biomappings.sssom.tsv -o filtered.sssom.tsv
0.79s user 0.11s system 281% cpu 0.818 total
```

Reply:
Which version of sssom-py?

Reply:
0.3.41.

Latest 0.4.0 is better but still decidedly not impressive:

```sh
$ time sssom parse -I tsv biomappings.sssom.tsv -m biomappings.sssom.yml -o combined.sssom.tsv
/Users/dpg44/Library/Python/3.11/lib/python/site-packages/sssom/util.py:164: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  non_double_cols.replace(np.nan, "", inplace=True)
14.53s user 0.30s system 117% cpu 13.239 total
```

~15s to catenate the metadata and the TSV… meh.


```sh
$ time sssom dosql -Q "SELECT * FROM df1 WHERE (subject_id LIKE 'UBERON:%' OR object_id LIKE 'UBERON:%')" -o filtered.sssom.tsv combined.sssom.tsv
/Users/dpg44/Library/Python/3.11/lib/python/site-packages/sssom/util.py:164: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  non_double_cols.replace(np.nan, "", inplace=True)
17.10s user 0.33s system 127% cpu 19.835 total
```

~20s for the filtering step.

Overall a ~2⨉ performance increase. I applaud the `sssom-py` developers for achieving that with Python, but when you start from such abysmal performances to begin with, even a 2⨉ increase is not great. :(

Reply:
Fair enough, although for a pipeline that runs 1 hour 15 seconds is not really that much of a deal in my humble view :P No worries, ignore me.

Reply:
Yes but you are trading an opportunity for standardising a workflow using a hack; there is a cost in that too; for example, there is a risk that the two pieces "don't fit together" (curie map is outdated etc), and you would never know. Its also less maintainable because there are 100 ways of doing this, and everyone could be doing it slightly differently creating cognitive overhead. I am not saying your judgement is wrong, I am just saying it that I would not consider it right to say that "runtime" is the only factor, and there are no tradeoffs.

Reply:
1. You can't complain that people are using Unix-level hacks to manipulate SSSOM file when you specifically designed the format to he amenable to this kind of hacks (quoting the SSSOM spec: “Additionally, it is a simple unix-level or language-level operation to filter these as a pre-processing in a robust fashion.”)

2. It's a _temporary_ hack until the next ODK version is available. As I said above, the next ODK will natively include my `sssom-cli` tool, which will allow to perform the desired operations (catenating the metadata and the mappings themselves, and filtering the non-Uberon mappings) in a completely “non-hacky” way in 1/20th of the time needed by `sssom-py`. In fact I _could_ have opted to manually download `sssom-cli` in this PR to use it straigh away (that would still have been faster than using `sssom-py`, even factoring on the downloading time...), but I decided it was not worth it.